app/models/payment_confirmation.rb

```diff
@@ -0,0 +1,87 @@
+class PaymentConfirmation
+  attr_reader :payroll_run, :csv_file, :updated_claim_ids
+  attr_accessor :errors
+
+  def initialize(payroll_run, csv_file)
+    @payroll_run = payroll_run
+    @errors = []
+    @csv_file = csv_file
+    @line_number = 1
+    @updated_claim_ids = Set.new
+  end
+
+  def ingest
+    return unless valid?
+
+    ActiveRecord::Base.transaction do
+      csv.body.each do |row|
+        @line_number += 1
+        payment = fetch_payment_by_reference(row["Claim ID"])
+        update_payment_fields(payment, row) unless payment.nil?
+      end
+
+      valid? || (raise ActiveRecord::Rollback)
+    end
+  end
+
+  def valid?
```

Comment:
I think it makes sense to do all of the nondestructive stuff in the initializer. I'll change that. 

Looking at the whole class again, though, I'm not really sure if the contract for this class is very clear. Do you think it makes sense that the return value of `valid?` can change after calling `ingest`?

Reply:
I think it's OK as calling ingest will be expected to mutate the state. Is it perhaps the name of the method? Would `successful?` make more sense? i.e. to indicate the outcome of the ingest?

Reply:
I think that of the two, `valid?` probably makes more sense, since we are able to call it both before and after `ingest?` is called (although we do not in practice). I'm probably giving it more thought than it deserves; will leave it as is.

Reply:
As discussed, I'll get rid of `valid?` and just use `errors`.